tsdb/mapper.go

```diff
@@ -88,6 +88,12 @@ func (lm *SelectMapper) Open() error {
 		return lm.remote.Open()
 	}
 
+	// This can happen when a shard has been assigned to this node but we have not
+	// written to it so it may not exist yet.
+	if lm.shard == nil {
+		return nil
+	}
+
```

Comment:
Probably need to add the same check here:  https://github.com/influxdb/influxdb/blob/master/tsdb/show_measurements.go#L144